docs/content/v0.20/release.md

```diff
@@ -0,0 +1,60 @@
+# Making KubeStellar Releases
+
+This document defines how releases of the KubeStellar repository are made.
+
+See the associated [packaging and delivery doc](packaging.md) for some
+clues about the problem.
+
+## Goals and limitations
+
+The release process has the following goals.
+
+- A release is identified using [semantic versioning](semver.org). This means that the associated semantics are followed, in terms of what sort of changes to the repo require what sort of changes to the release identifier.
+- A user can pick up and use a given existing release without being perturbed by on-going contributor work. A release is an immutable thing.
+- A release with a given semver identifier is built from a commit of this Git repository tagged with a tag whose name is "v" followed by the release identifier.
+- There is a concept of a release branch, with a name following the pattern `release-$major.$minor`. This allows contributors to do maintenance work on older release branches. This also allows a user to track work along such a line.
+- The contents of `main` always work. This includes passing CI tests. This includes documentation being accurate. We allow point-in-time specific documentation, such as a document that says "Here is how to use release 1.2.3" --- which would refer to a release made in the past. We do not require the documentation in `main` to document all releases.
+- A git tag is immutable. Once associated with a given Git commit, that association is not changed later.
+- We do not put self-references into Git. For example, making release `1.2.3` doe not require changing any file in Git to have the string `1.2.3` in it.
+
+We have the following limitations.
+
+- The only way to publish artifacts (broadly construed, not (necessarily) GitHub "release artifacts") is to make a release.
```

Comment:
This seems a technical limitation  and it also add a lot of complexity to testing and naming.

Reply:
I assume you the technical limitation is around GoReleaser, right ? As we have seen, we should avoid manual publishing to ghcr.io because of the ownership/permissions issues. If we do not publish artifacts for testing via GoReleaser, we can still create GitHub workflows just for that purpose. We can still trigger the workflows based on a tag, but perhaps use a different pattern from that used for a release (we would need also to change the pattern that trigger the workflow invoking goreleaser)

Reply:
It is not just because of goreleaser. You would do better to ask if it is because of the authorization problem at ghcr.io. But the answer is the "release tests". How else would we test a release without making a release?

Reply:
@pdettori: put another way: we need a process to publish artifacts for a release. I am trying to not multiply entities needlessly.